Networking/Networking/Remote/CouponsRemote.swift

```diff
@@ -255,6 +260,65 @@ public final class CouponsRemote: Remote, CouponsRemoteProtocol {
             }
         })
     }
+
+    // MARK: - Load most active coupons
+
+    /// Loads top 3 most active coupons based on order count within the specified time range and site ID.
+    ///
+    /// - Parameters:
+    ///     - siteID: The ID of the  site from which we'll fetch the coupons report.
+    ///     - from: The start of the date range for which we'll fetch the coupons report. Sending `nil`fetches all time info.
+    ///     - to: The end of the date range until which we'll fetch the coupons report. Sending `nil`fetches all time info.
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    public func loadMostActiveCoupons(for siteID: Int64,
+                                      from startDate: Date? = nil,
+                                      to endDate: Date? = nil,
```

Comment:
Question:

Is the API always returning all time info if either start or end date is nil?

If yes, in terms of using `nil` to signify "fetching all time info", would it be clearer if we make a `DateRange` struct similar to [OrderDateRangeFilter](https://github.com/woocommerce/woocommerce-ios/blob/3050bcac5ca81fff1c36db16283ec58822d74bd9/Yosemite/Yosemite/Model/Orders/OrderDateRangeFilter.swift#L5)?

Then we can just have one parameter that can be nil instead of two.

Reply:
Good question, Hafiz!

Internal - p1715321969895879/1715059400.957489-slack-C03L1NF1EA3

We wanted to reuse the `StatsTimeRangeV4` for coupons cards as well, but it makes the UI date-picking logic cumbersome. 

I introduced a separate enum in https://github.com/woocommerce/woocommerce-ios/pull/12699/commits/5b9b7f24d67d7b4f7e9d357ff7fd7a4159f47df3.

Reply:
I kept the `MostActiveCouponsTimeRange` in the Yosemite layer just like `StatsTimeRangeV4` as we are only sending dates and not sending enums/raw values in network request. i.e. Networking layer doesn't have to know about the `MostActiveCouponsTimeRange`.